lib/setup.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+const Assert = require('assert');
+const CloudApi = require('webconsole-cloudapi-client');
+
+const setupCloudApi = ({ keyId, key, apiBaseUrl }) => {
+  return ({ auth, log }) => {
+    return new CloudApi({
+      token: auth.credentials && auth.credentials.token,
+      url: apiBaseUrl,
+      keyId,
+      key,
+      log
+    });
+  };
+};
+
+const preResolve = (cloudapi) => {
+  return (root, args, request) => {
+    if (request.route.settings.auth === false) {
+      return;
+    }
+
+    return cloudapi({ auth: request.auth, log: request.log.bind(request) });
+  };
+};
+
+const postAuth = (cloudapi) => {
+  return (request, h) => {
+    if (request.route.settings.auth === false) {
+      return h.continue;
+    }
+
+    request.plugins.cloudapi = cloudapi({ auth: request.auth, log: request.log.bind(request) });
+
+    return h.continue;
+  };
+};
+
+const graphqlHandler = function (route, options) {
+  Assert(typeof options.method === 'function', 'method must be a function');
+
+  return function (request, h) {
+    const fetch = request.plugins.cloudapi.fetch.bind(request.plugins.cloudapi);
+    return options.method(fetch, request.payload, request);
+  };
+};
+
+module.exports = {

```

Comment:
we should just export directly on each of the functions instead of an export at the bottom.

Reply:
Why is this preferred? Just curious.